R/rdeephaven/R/agg_ops_wrapper.R

```diff
@@ -1,6 +1,134 @@
+#' `agg` functions are used to perform aggregation calculations on grouped data by passing them to `agg_by()` or
+#' `agg_all_by()`. These functions are _generators_, meaning they return _functions_ that the Deephaven engine knows
+#' how to interpret. We call the functions that they return [`AggOp`][AggOp]s. These `AggOp`s are not R-level functions,
+#' but Deephaven-specific data types that perform all of the intensive calculations. Here is a list of all `agg` functions
+#' available in Deephaven:
+#'
+#' - [`agg_first()`][agg_first]
+#' - [`agg_last()`][agg_last]
+#' - [`agg_min()`][agg_min]
+#' - [`agg_max()`][agg_max]
+#' - [`agg_sum()`][agg_sum]
+#' - [`agg_abs_sum()`][agg_abs_sum]
+#' - [`agg_avg()`][agg_avg]
+#' - [`agg_w_avg()`][agg_w_avg]
+#' - [`agg_median()`][agg_median]
+#' - [`agg_var()`][agg_var]
+#' - [`agg_std()`][agg_std]
+#' - [`agg_percentile()`][agg_percentile]
+#' - [`agg_count()`][agg_count]
+#'
+#' For more details on each aggregation function, click on one of the methods above or see the reference documentation
+#' by running `?agg_first`, `?agg_last`, etc.
+#'
+#' @examples
+#' \dontrun{
+#' library(rdeephaven)
+#'
+#' # connecting to Deephaven server
+#' client <- Client$new("localhost:10000", auth_type="psk", auth_token="my_secret_token")
+#'
+#' # create data frame, push to server, retrieve TableHandle
+#' df <- data.frame(
+#'   X = c("A", "B", "A", "C", "B", "A", "B", "B", "C"),
+#'   Y = c("M", "N", "O", "N", "P", "M", "O", "P", "M"),
+#'   Number1 = c(100, -44, 49, 11, -66, 50, 29, 18, -70),
+#'   Number2 = c(-55, 76, 20, 130, 230, -50, 73, 137, 214)
+#' )
+#' th <- client$import_table(df)
+#'
+#' # get first and last elements of each column
+#' th1 <- th$
+#'   agg_by(agg_first(c("XFirst = X", "YFirst = Y", "Number1First = Number1", "Number2First = Number2")),
+#'          agg_last(c("XLast = X", "YLast = Y", "Number1Last = Number1", "Number2Last = Number2")))
+#'
+#' # compute mean and standard deviation of Number1 and Number2, grouped by X
+#' th2 <- th$
+#'   agg_by(
+#'     c(agg_avg(c("Number1Avg = Number1", "Number2Avg = Number2")),
+#'       agg_std(c("Number1Std = Number1", "Number2Std = Number2"))),
+#'     by="X")
+#'
+#' # compute maximum of all non-grouping columns, grouped by X and Y
+#' th3 <- th$
+#'   agg_all_by(agg_max(), by=c("X", "Y"))
+#'
+#' # compute minimum and maximum of Number1 and Number2 respectively grouped by Y
+#' th4 <- th$
+#'   agg_by(
+#'     c(agg_min("Number1Min = Number1"),
+#'       agg_max("Number2Max = Number2")),
+#'     by="Y")
+#'
+#' client$close()
+#' }
+#'
+NULL
+
+
+#' Name AggOp
+#' @title Deephaven AggOps
+#' @md
 #' @description
-#' An AggOp represents an aggregation operation that can be passed to `agg_by()` or `agg_all_by()`.
-#' Note that AggOps should not be instantiated directly by user code, but rather by provided agg_* functions.
+#' An `AggOp` is the return type of one of Deephaven's [`agg`][AggBy] functions. It is a function that performs the
```

Comment:
Should indicate that these are "deephaven" functions and not general R functions.